### PR TITLE
Add typecheck linter for typescript

### PR DESCRIPTION
**Pull request by mkusher**
Solves #20 

- Linters name: `typecheck`
- Required npm package: `ts-type-checker`(https://www.npmjs.com/package/ts-type-checker)

It uses my simple wrapper around typescript compiler . `npm i -g ts-type-checker` would be enough.
typecheck.vim is almost a copy of tslint.vim.

### CODE REVIEW

File: ale_linters/typescript/typecheck.vim [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        \})
+    endfor
+
+    return l:output
+endfunction
+
+call ale#linter#Define('typescript', {
+\   'name': 'typecheck',
+\   'executable': 'typecheck',
+\   'command': 'typecheck %s',
```
Comment (w0rp) [inline ale_linters/typescript/typecheck.vim]: That will be running the command on the file as it is on disk, not the text you are editing as you type. You'll either need to use an option to use `stdin` data for the file, or use the stdin_wrapper file, probably with an argument to include the directory you are editing the file in or something similar.

File: ale_linters/typescript/typecheck.vim [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    " hello.ts[7, 41]: Property 'a' does not exist on type 'A'
+    " hello.ts[16, 7]: Type 'A' is not assignable to type 'B'
+    "
+    let l:pattern = '.\+.ts\[\(\d\+\), \(\d\+\)\]: \(.\+\)'
+    let l:output = []
+
+    for l:line in a:lines
+        let l:match = matchlist(l:line, l:pattern)
+        echom len(l:match)
+        echom join(l:match, ", ")
```
Comment (w0rp) [inline ale_linters/typescript/typecheck.vim]: You left some echo lines here.

Comment (mkusher) [inline ale_linters/typescript/typecheck.vim]: oh, my fault

File: ale_linters/typescript/typecheck.vim (hunk 1 above)
Comment (mkusher) [inline ale_linters/typescript/typecheck.vim]: no, typecheck uses stdin by default

Comment (w0rp) [inline ale_linters/typescript/typecheck.vim]: Does it still use stdin if a filename is given? The `'%s'` here will be replaced with the filename.

Comment (mkusher) [inline ale_linters/typescript/typecheck.vim]: yes, typecheck needs file name to find tsconfig.json, but it will read file contents from stdin

Comment (w0rp) [inline ale_linters/typescript/typecheck.vim]: Ah, sounds good to me, then. Sounds similar to the `--stdin-filename` option for `eslint`, but as a positional argument.

**Comment by w0rp** [comment]
I think this is good. I'll let Travis CI look at this. We might find that it could be a bit slow, but it's certainly better than nothing. If it delays other checks, you can always turn it off. If someone can implement something using HTTP and the test server, I'd also accept that.

**Pull request merged**